lld/ELF/InputSection.cpp

```diff
@@ -917,8 +925,7 @@ void InputSection::relocateNonAlloc(uint8_t *buf, ArrayRef<RelTy> rels) {
     if (expr == R_NONE)
       continue;
 
-    if (tombstone ||
-        (isDebug && (type == target.symbolicRel || expr == R_DTPREL))) {
+    if (tombstoneValueToUse) {
```

Comment:
I think
`if (!sym.getOutputSection() || (ds && ds->folded && !isDebugLine))` should be the outer `if` and the tombstone logic should be the inner `if`.